overlord/devicestate/devicestate.go

```diff
@@ -185,8 +185,9 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, flags sn
 		return fmt.Errorf("cannot install %s snap on classic if not requested by the model", kind)
 	}
 
-	if snapInfo.Name() != expectedName {
-		return fmt.Errorf("cannot install %s %q, model assertion requests %q", kind, snapInfo.Name(), expectedName)
+	// TODO parallel-install: use store name
```

Comment:
we will need to add checks in the model assertion code to check for valid store snap names, before it was not needed because the use/installing of the snaps would check for them, but now we don't want gadget_foo or kernel_foo to be used

from that POV we want the instance name (which then must == store name) here

Reply:
probably a TODO inside https://github.com/snapcore/snapd/blob/master/asserts/device_asserts.go#L168
would be good